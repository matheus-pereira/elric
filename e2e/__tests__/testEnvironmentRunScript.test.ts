/**
 * Copyright (c) Facebook, Inc. and its affiliates. All Rights Reserved.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 */

import {resolve, sep} from 'path';
import runelric from '../runelric';

it('throw error if test env does not have getVmContext', () => {
  const DIR = resolve(__dirname, '../test-environment-run-script');
  const {exitCode, stderr} = runelric(DIR);

  expect(stderr.replace(`${DIR}${sep}`, '<rootDir>/')).toMatchSnapshot();
  expect(exitCode).toBe(1);
});
