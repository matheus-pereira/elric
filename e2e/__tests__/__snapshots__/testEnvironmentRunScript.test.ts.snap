// elric Snapshot v1, https://goo.gl/fbAQLP

exports[`throw error if test env does not have getVmContext 1`] = `"Test environment found at \\"<rootDir>/EnvUsingRunScript.js\\" does not export a \\"getVmContext\\" method, which is mandatory from elric 27. This method is a replacement for \\"runScript\\"."`;
